pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/DatabaseTestUtil.java

```diff
@@ -0,0 +1,103 @@
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.constants.ConfigFile;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.exception.DDPException;
+
+@Slf4j
+public class DatabaseTestUtil {
+    public static final String TESTING_ROOT_USERNAME = "root";
+    public static final String TESTING_ROOT_PASSWORD = "test";
+
+    public static final String MYSQL_VERSION = "mysql:5.7";
+
+    // classpath location for init script for disposable mysql test db
+    public static final String TEST_DB_MYSQL_CONF_DIR = "testdbs";
+
+    private DatabaseTestUtil() {}
+
+    /**
+     * Set up a database for testing based on supplied configuration and initialize TransactionWrapper
+     */
+    public static void initDbConnection(Config cfg, TransactionWrapper.DB db) {
+        int maxConnections = cfg.getInt(ConfigFile.NUM_POOLED_CONNECTIONS);
+        String dbUrl = cfg.getString(db.getDbUrlConfigKey());
+        log.info("Initializing db pool");
+        TransactionWrapper.reset();
+        TransactionWrapper.init(new TransactionWrapper.DbConfiguration(db, maxConnections, dbUrl));
+
+        LiquibaseUtil.runLiquibase(dbUrl, db, LiquibaseUtil.LIQUIBASE_TEST_CONTEXT);
+    }
+
+    public static void closeDbConnection() {
+        TransactionWrapper.reset();
+    }
+
+    public static Optional<String> parseDbUrlPassword(String dbUrl) {
+        try {
+            URI url = new URI(dbUrl.replace("jdbc:", ""));
+            String[] params = url.getQuery().split("&");
+            for (String param : params) {
+                String[] parts = param.split("=");
+                if ("password".equals(parts[0])) {
+                    return Optional.of(parts[1]);
+                }
+            }
+        } catch (URISyntaxException e) {
+            log.error("invalid db url", e);
+            return Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Creates a new Config by overwriting the database passwords using
+     * the given values
+     */
+    public static Config overrideConfigFileDbPasswords(String newPassword) {
+        Config modifiedConfig = ConfigManager.getInstance().getConfig();
+        String[] dbUrlKeys = new String[] {ConfigFile.DB_URL, ConfigFile.HOUSEKEEPING_DB_URL};
+        for (String dbUrlKey : dbUrlKeys) {
+            if (modifiedConfig.hasPath(dbUrlKey)) {
+                String dbUrlValue = modifiedConfig.getString(dbUrlKey);
+                String oldPassword = parseDbUrlPassword(dbUrlValue).orElse(null);
+                if (oldPassword != null) {
+                    String newDbUrl = dbUrlValue.replace("&password=" + oldPassword, "&password=" + newPassword);
+                    modifiedConfig = modifiedConfig.withValue(dbUrlKey, ConfigValueFactory.fromAnyRef(newDbUrl, "Overridden value"));
+                } else {
+                    throw new DDPException("Could not find old password in config file for db url key: " + dbUrlKey);
```

Comment:
Could this leak a secret?  Would it be better to not return the dbUrlKey and instead a path to the file that's being used?

Reply:
Erf. This file is mostly pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/util/MySqlTestContainerUtil.java moved into ddp-common, with the addition of the initializeTestDbs method.
I did not see a good way to preserve the git history, but if you know of a way pls share it.
In any case, I did not examine this code very carefully, but it might be ok?
This exception will only be thrown from a test, and providing the key helps the tester quickly identify the value they need to fix (e.g. `dbUrl`).
But if you feel there is a security concern we could simply refer to the key indirectly instead of providing it (e.g. 'the database URL key').
What do you think?